if (!require("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install("clusterProfiler")
BiocManager::install("GreyListChIP")

BiocManager::install("DiffBind", force = TRUE)

browseVignettes("DiffBind")
library(DiffBind)
library(ggplot2)
library()
getwd()
setwd(system.file('extra',package='DiffBind'))

tmpdir <- tempdir()
url <- 'https://content.cruk.cam.ac.uk/bioinformatics/software/DiffBind/DiffBind_vignette_data.tar.gz'
file <- basename(url)
options(timeout=600)
download.file(url, file.path(tmpdir,file))
untar(file.path(tmpdir,file), exdir = tmpdir )
setwd(file.path(tmpdir,"DiffBind_Vignette"))

# analysis in one step
tamoxifen <- dba.analyze("tamoxifen.csv")

tamoxifen.DB <- dba.report(tamoxifen)

# analysis in multiple steps
tamoxifen <- dba(sampleSheet="tamoxifen.csv") %>%
  + dba.blacklist() %>%
  + dba.count() %>%
  + dba.normalize() %>%
  + dba.contrast() %>%
  + dba.analyze()

# Reading in of peakset
samples <- read.csv(file.path(system.file("extra", package="DiffBind"), "tamoxifen.csv"))
names(samples)
samples
tamoxifen <- dba(sampleSheet="tamoxifen.csv", dir=system.file("extra", package="DiffBind"))

# Alternatively, the previously read-in sample sheet could be used directly to create the DBA
object:tamoxifen <- dba(sampleSheet=samples)

# metadata associated with this object can be displayed simply as
tamoxifen

# Using the data from the peak calls, a correlation heatmap can be generated
plot(tamoxifen)
## can also be generated by
dba.plotHeatmap(tamoxifen)

# Counting reads
tamoxifen <- dba.count(tamoxifen)

# After the dba.count call, the DBA object can be examined:
tamoxifen

# number of reads that overlap a consensus peak
info <- dba.show(tamoxifen)
libsizes <- cbind(LibReads=info$Reads, FRiP=info$FRiP, PeakReads=round(info$Reads * info$FRiP))
rownames(libsizes) <- info$ID
libsizes

# new correlation heatmap and PCA plot based on the count scores (using default normalization parameters)
plot(tamoxifen)
dba.plotPCA(tamoxifen,DBA_TISSUE,label=DBA_CONDITION)

# Normalizing the data
## By default, the data are normalized based on sequencing depth.
tamoxifen <- dba.normalize(tamoxifen)
## The details of the normalization can be examined
norm <- dba.normalize(tamoxifen, bRetrieve=TRUE)
norm
## The default library-size based methods result in all the library sizes being normalized to be the same (the mean library size)
normlibs <- cbind(FullLibSize=norm$lib.sizes, NormFacs=norm$norm.factors, NormLibSize=round(norm$lib.sizes/norm$norm.factors))
rownames(normlibs) <- info$ID
normlibs

# Before running the differential analysis, establishing a model design and contrast
tamoxifen <- dba.contrast(tamoxifen, reorderMeta=list(Condition="Responsive"))
tamoxifen

# Performing the differential analysis
tamoxifen <- dba.analyze(tamoxifen)
dba.show(tamoxifen, bContrasts=TRUE)

# heatmap and other plots correlating only the differentially bound sites identified by the analysis
plot(tamoxifen, contrast=1)
dba.plotPCA(tamoxifen, contrast=1, label=DBA_TISSUE)
dba.plotMA(tamoxifen)
dba.plotVolcano(tamoxifen)

# how read distributions differ between classes of binding sites
sum(tamoxifen.DB$Fold<0)
sum(tamoxifen.DB$Fold>0)

# how are reads distributed amongst the different classes of differentially bound sites and sample groups, can be more clearly seen using a boxplot
pvals <- dba.plotBox(tamoxifen)

#binding affinity heatmap, showing the read scores for some or all of the binding sites
hmap <- colorRampPalette(c("red", "black", "green"))(n = 13)
readscores <- dba.plotHeatmap(tamoxifen, contrast=1, correlations=FALSE, scale="row", colScheme = hmap)

# Retrieving the differentially bound sites as a GRanges object
tamoxifen.DB <- dba.report(tamoxifen)
tamoxifen.DB


